functions/index.js

```diff
@@ -26,25 +26,42 @@ exports.helloWorld = functions.https.onRequest((request, response) => {
 exports.pullDataFromLocalCSVFileTEST = functions.https.onRequest((request, response) => {
   let fileContent="";
   let pathToFile="";
-  pathToFile = request.body.pathToFile
+  let clientOrGoalCSV = "";
+  pathToFile = request.body.pathToFile;
+  clientOrGoalCSV = request.body.clientOrGoalCSV;
   console.log('Extracting data from the following file: ' + JSON.stringify(pathToFile));
   let csvzip = new AdmZip(pathToFile);
   let zipEntries = csvzip.getEntries();
   if (zipEntries.length > 0) {
       console.log('Found ' + zipEntries.length + ' entry in the zip file');
       fileContent += csvzip.readAsText(zipEntries[0]);
-}
-  parseCSVAndSaveToFireStore (fileContent);
+      }
+  switch (clientOrGoalCSV) {
+    case 'client':
+      parseClientCSVAndSaveToFireStore (fileContent);
+      break;
+    case 'goal':
+      parseGoalCSVAndSaveToFireStore (fileContent);
+      break;
+    default:
+      throw ("Must provide argument <client|goal>");
```

Comment:
```suggestion
      response
        .type('application/json')
        .status(409)
        .send({status:409, message: "Missing required param clientOrGoalCSV of either 'client ' or 'goal'" });
```
Ideally apis should always return a response instead of throwing an error.  This way developers (including ourselves) can quickly see feedback when we go astray instead of spending time looking through logs.  This snippit is a bit of a guess [based on the docs](https://expressjs.com/en/api.html#res.send), but give it a try and see if you get the expected response.